### PR TITLE
fix(ForeignKeyMetadata): default for onUpdate/onDelete is RESTRICT

### PR DESCRIPTION
**Pull request by adamlacombe**

### CODE REVIEW

**Comment by vlapo** [comment]
Hey @adamlacombe thank you for your PR. I love to see new contributors coming to typeorm :)

But please update description of change (why you want this change) and also check failed tests.

File: src/metadata/ForeignKeyMetadata.ts [hunk 1]
```diff
@@ -87,8 +88,14 @@ export class ForeignKeyMetadata {
...
         this.columns = options.columns;
         this.referencedColumns = options.referencedColumns;
-        this.onDelete = options.onDelete || "NO ACTION";
-        this.onUpdate = options.onUpdate || "NO ACTION";
+    
+        /**
+         * "For an ON DELETE or ON UPDATE that is not specified, the default action is always RESTRICT."
+         * https://dev.mysql.com/doc/refman/8.0/en/create-table-foreign-keys.html
+         */
+        this.onDelete = options.onDelete || (this.entityMetadata.connection.driver instanceof MysqlDriver) ? "RESTRICT" : "NO ACTION";
```
Comment (adamlacombe) [inline src/metadata/ForeignKeyMetadata.ts]: Not sure why its throwing [this error](https://travis-ci.org/typeorm/typeorm/jobs/523736835#L1665)

This condition seems to be failing for some reason. Any suggestions? 
```typescript 
(this.entityMetadata.connection.driver instanceof MysqlDriver) 
```

**Comment by pleerock** [comment]
Please provide a detailed description of what you are trying to do and why

**Comment by imnotjames** [comment]
Still have failed tests here.

Can you rebase, fix the test failures, and update the description of the PR with what you're trying to do?

**Comment by imnotjames** [comment]
Thanks for the contribution but without any feedback / activity in ~2 years I'll be closing this out.  Apologies if there's an open issue for this, it's not linked here.

**Pull request closed without merging**